nipst/post.go

```diff
@@ -46,23 +47,29 @@ type PostClient struct {
 var _ PostProverClient = (*PostClient)(nil)
 
 func NewPostClient(cfg *config.Config) *PostClient {
-	return &PostClient{cfg, nil, shared.DisabledLogger{}}
+	return &PostClient{nil, cfg, nil, shared.DisabledLogger{}}
```

Comment:
In both of `SetParams` usages you know `id` beforehand. Any reason for not setting it here, and so to avoid need to assert that `id != nil` on each call?
You can also remove `SetParams` entirely, and provide the `datadir`/`space` args into `initialize`/`execute`. I think it will be less prone to bugs.